dispatcher.go

```diff
@@ -75,11 +76,13 @@ func NewDispatcher(cfg Config) *Dispatcher {
 		panic("yarpc.NewDispatcher expects a valid service name: " + err.Error())
 	}
 
+	once := lifecycle.NewOnce()
 	logger := cfg.Logging.logger(cfg.Name)
 	extractor := cfg.Logging.extractor()
 
 	registry, stopPush := cfg.Metrics.registry(cfg.Name, logger)
 	cfg = addObservingMiddleware(cfg, registry, logger, extractor)
+	cfg = addIsRunningMiddleware(cfg, once.IsRunning)
```

Comment:
nit: Can we add all the internal middleware in the same function
(`addInternalMiddleware`?) so that the order in which the middleware is
executed is more obvious.

    cfg.InboundMiddleware.Unary = UnaryInboundMiddleware(
        inboundIsRunning,
        observer,
        cfg.InboundMiddleware.Unary,
    )

Reply:
Done